### ISSUE TITLE
New predict API for multiple outputs

### ISSUE THREAD

**Issue by snakeztc** [fchollet/keras#2422]
Before the new update, if all the output layers are named and in graph model, calling predict(X) will return a dictionary. The new API seems can only return a list of prediction. Will the predict be able to output dictionary in the future? There are some inconsistence because the fit() function can take in "y" as a dict while the predict() cannot. Thanks!

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [x] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by AntonOsika** [comment]
If you have named your output layers this code should convert predictions to a dictionary:
```python
p = model.predict(x)
p = {out.name.split(':')[0]: p[i] for i, out in enumerate(model.outputs)}
```

**Comment by nbro** [comment]
@AntonOsika But is the order of the predictions in the list `p` ensured to be the order of the pairs in the dictionary passed to the loss and metrics? Dictionaries do not necessarily ensure that the order of the elements is constant.
